# swagger-client
This is PeptideMatch OpenAPI.

This Python package is automatically generated by the [Swagger Codegen](https://github.com/swagger-api/swagger-codegen) project:

- API version: 2.0.0
- Package version: 1.0.0
- Build package: io.swagger.codegen.languages.PythonClientCodegen

## Requirements.

Python 2.7 and 3.4+

## Installation & Usage
### pip install

If the python package is hosted on Github, you can install directly from Github

```sh
pip install git+https://github.com//.git
```
(you may need to run `pip` with root permission: `sudo pip install git+https://github.com//.git`)

Then import the package:
```python
import swagger_client 
```

### Setuptools

Install via [Setuptools](http://pypi.python.org/pypi/setuptools).

```sh
python setup.py install --user
```
(or `sudo python setup.py install` to install the package for all users)

Then import the package:
```python
import swagger_client
```

## Getting Started

Please follow the [installation procedure](#installation--usage) and then run the following:

```python
from __future__ import print_function
import time
import swagger_client
from swagger_client.rest import ApiException
from pprint import pprint
# create an instance of the API class
api_instance = swagger_client.PeptideMatchAPI20Api()
peptides = 'AAVEEGIVLGGGCALLR,SVQYDDVPEYK' # str | A list of comma-separated peptide sequences (up to 100). Each sequence consists of 3 or more amino acids. (default to AAVEEGIVLGGGCALLR,SVQYDDVPEYK)
taxonids = '9606,10090' # str | A list fo comma-separated NCBI taxonomy IDs. (optional) (default to 9606,10090)
swissprot = true # bool | Only search SwissProt protein sequences. (optional) (default to true)
isoform = true # bool | Include isforms. (optional) (default to true)
uniref100 = false # bool | Only search UniRef100 protein sequences. (optional) (default to false)
leqi = false # bool | Treat Leucine (L) and Isoleucine (I) equivalent. (optional) (default to false)
offset = 0 # int | Off set, page starting point, with default value 0. (optional) (default to 0)
size = 100 # int | Page size with default value 100. When page size is -1, it returns all records and offset will be ignored. (optional) (default to 100)

try:
    # Do peptide match using GET method.
    api_response = api_instance.match_get_get(peptides, taxonids=taxonids, swissprot=swissprot, isoform=isoform, uniref100=uniref100, leqi=leqi, offset=offset, size=size)
    pprint(api_response)
except ApiException as e:
    print("Exception when calling PeptideMatchAPI20Api->match_get_get: %s\n" % e)

```

## Documentation for API Endpoints

All URIs are relative to *http://research.bioinformatics.udel.edu/peptidematchapi2*

Class | Method | HTTP request | Description
------------ | ------------- | ------------- | -------------
*PeptideMatchAPI20Api* | [**match_get_get**](docs/PeptideMatchAPI20Api.md#match_get_get) | **GET** /match_get | Do peptide match using GET method.
*PeptideMatchAPI20Api* | [**match_post_post**](docs/PeptideMatchAPI20Api.md#match_post_post) | **POST** /match_post | Do peptide match using POST method.


## Documentation For Models

 - [Error](docs/Error.md)
 - [Protein](docs/Protein.md)
 - [ProteinMatchRange](docs/ProteinMatchRange.md)
 - [ProteinMatchingPeptide](docs/ProteinMatchingPeptide.md)
 - [Report](docs/Report.md)
 - [ReportResults](docs/ReportResults.md)
 - [ReportSearchParameters](docs/ReportSearchParameters.md)


## Documentation For Authorization

 All endpoints do not require authorization.


## Author

chenc@udel.edu

